netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -165,14 +166,20 @@ public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers) {
   public Runnable start(Listener transportListener) {
     lifecycleManager = new ClientTransportLifecycleManager(
```

Comment:
did you forget to replace the `lifecycleManager ` field with a local variable? Ditto for the `keepaliveManager` field if it is not exposed to test.